testworld/signature_test.go

```diff
@@ -0,0 +1,165 @@
+	assert.Equal(t, 1, len(dm.Signatures()))
+
+	signatures, signatureErrors, err := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.Nil(t, signatureErrors)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(signatures))
+}
+
+func TestHost_FakedSignature(t *testing.T) {
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	actxh := testingconfig.CreateAccountContext(t, alice.host.config)
+	ectxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	// Get PublicKey and PrivateKey
+	publicKey, privateKey := GetSigningKeyPair(t, alice.host.idService, alice.id, actxh)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, publicKey, privateKey)
+
+	signatures, signatureErrors, _ := eve.host.p2pClient.GetSignaturesForDocument(ectxh, dm)
+	assert.Error(t, signatureErrors[0], "Signature verification failed error")
+	assert.Equal(t, 0, len(signatures))
+}
+
+func TestHost_RevokedSigningKey(t *testing.T) {
+	// Hosts
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ctxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	// Get PublicKey and PrivateKey
+	publicKey, privateKey := GetSigningKeyPair(t, eve.host.idService, eve.id, ctxh)
+
+	// Revoke Key
+	key, err := utils.SliceToByte32(publicKey)
+	assert.NoError(t, err)
+	RevokeSigningKey(t, eve.host.idService, key, eve.id, ctxh)
+
+	collaborators := [][]byte{bob.id[:]}
+	dm := createCDWithEmbeddedPO(t, collaborators, eve.id, publicKey, privateKey)
+
+	signatures, signatureErrors, err := eve.host.p2pClient.GetSignaturesForDocument(ctxh, dm)
+	assert.NoError(t, err)
+	assert.Error(t, signatureErrors[0], "Signature verification failed error")
+	assert.Equal(t, 0, len(signatures))
+}
+
+// Helper Methods
+func createCDWithEmbeddedPO(t *testing.T, collaborators [][]byte, identityDID identity.DID, publicKey []byte, privateKey []byte) documents.Model {
+	payload := testingdocuments.CreatePOPayload()
+	var cs []string
+	for _, c := range collaborators {
+		cs = append(cs, hexutil.Encode(c))
+	}
+	payload.Collaborators = cs
+
+	po := new(purchaseorder.PurchaseOrder)
+	err := po.InitPurchaseOrderInput(payload, identityDID.String())
+	assert.NoError(t, err)
+
+	_, err = po.CalculateDataRoot()
+	assert.NoError(t, err)
+
+	sr, err := po.CalculateSigningRoot()
+	assert.NoError(t, err)
+
+	s, err := crypto.SignMessage(privateKey, sr, crypto.CurveSecp256K1)
+	assert.NoError(t, err)
+
+	sig := &coredocumentpb.Signature{
+		EntityId:  identityDID[:],
+		PublicKey: publicKey,
+		Signature: s,
+		Timestamp: utils.ToTimestamp(time.Now().UTC()),
+	}
+	po.AppendSignatures(sig)
+
+	_, err = po.CalculateDocumentRoot()
+	assert.NoError(t, err)
+
+	return po
+}
+
+func RevokeSigningKey(t *testing.T, idService identity.ServiceDID, key [32]byte, identityDID identity.DID, ctx context.Context) {
```

Comment:
`RevokeSigningKey` -> `RevokeKey` (since its not specific to a signing key here)